### PR TITLE
Merge client

### PR DESCRIPTION
**Pull request by rainbeam**
### What was wrong?

The [ethereum-tester-client](https://github.com/pipermerriam/ethereum-tester-client) was an external dependency of eth-testrpc, however it no longer made sense to keep the code out of this repo.
### How was it fixed?

The python source and its history have been extracted from ethereum-tester-client and merged into eth-testrpc.
#### Cute Animal Picture

rhino
#### TODO
- [x] fix failing filter tests (all in tests/filters/test_new_filter.py)
-  ~~remove now redundant code~~
  - ~~change rpc_client -> client~~
  - ~~endpoint tests~~

### CODE REVIEW

**Comment by rainbeam** [comment]
@pipermerriam: am I right in thinking that all instances of `rpc_client` can now be replaced with `client` in the tests?

**Comment by pipermerriam** [comment]
@rainbeam yes, feel free to do that.

**Comment by pipermerriam** [comment]
@rainbeam wait, no, that is not correct.  The `client` fixture interacts with it directly from python.  The `rpc_client` interacts with it over HTTP via the JSON-RPC server.  We want to keep this distinction in place.

File: tests/utility/test_str_and_bytes_utils.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ALL_ASCII_BINARY = b'\x00\x01\x02\x03\x04\x05\x06\x07\x08\t\n\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f !"#$%&\'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~\x7f\x80\x81\x82\x83\x84\x85\x86\x87\x88\x89\x8a\x8b\x8c\x8d\x8e\x8f\x90\x91\x92\x93\x94\x95\x96\x97\x98\x99\x9a\x9b\x9c\x9d\x9e\x9f\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa9\xaa\xab\xac\xad\xae\xaf\xb0\xb1\xb2\xb3\xb4\xb5\xb6\xb7\xb8\xb9\xba\xbb\xbc\xbd\xbe\xbf\xc0\xc1\xc2\xc3\xc4\xc5\xc6\xc7\xc8\xc9\xca\xcb\xcc\xcd\xce\xcf\xd0\xd1\xd2\xd3\xd4\xd5\xd6\xd7\xd8\xd9\xda\xdb\xdc\xdd\xde\xdf\xe0\xe1\xe2\xe3\xe4\xe5\xe6\xe7\xe8\xe9\xea\xeb\xec\xed\xee\xef\xf0\xf1\xf2\xf3\xf4\xf5\xf6\xf7\xf8\xf9\xfa\xfb\xfc\xfd\xfe\xff'
+ALL_ASCII_BYTEARRAY = bytearray(range(256))
+
+
+@pytest.mark.parametrize(
+    'input,expected',
+    (
+		(EMPTY_BINARY, EMPTY_TEXT),
```
Comment (rainbeam) [inline tests/utility/test_str_and_bytes_utils.py]: fixed in 6cd4310

**Comment by rainbeam** [comment]
Tests are all passing now. There is an error that comes up at the end though:

```
Exception ignored in: <bound method state.__del__ of <ethereum.tester.state object at 0x7f6a06f54828>>
Traceback (most recent call last):
  File "/home/rain/src/miniconda/envs/testrpc/lib/python3.5/site-packages/ethereum/tester.py", line 187, in __del__
AttributeError: 'NoneType' object has no attribute 'rmtree'
```

Not sure if you'd been seeing this before? I can see something similar to this on master.

**Comment by pipermerriam** [comment]
Interesting, that lends some possible insight as to where that error comes from.  Thanks for reporting.

In other news, this still has a WIP in the title.  Is that the case or is it ready for review and merge?

**Comment by rainbeam** [comment]
@pipermerriam: please review.

**Pull request merged**